src/sage/matroids/oriented_matroids/oriented_matroid.py

```diff
@@ -0,0 +1,737 @@
+            sage: M = OrientedMatroid([[1], [-1]], key='circuit')
+            sage: M.circuits()
+            [+: 0
+             -:
+             0: ,
+             +:
+             -: 0
+             0: ]
+        """
+        if hasattr(self, "_circuits"):
+            return self._circuits
+        raise NotImplementedError("circuits not implemented")
+
+    def cocircuits(self):
+        """
+        Return all cocircuits.
+        """
+        if hasattr(self, "_cocircuits"):
+            return self._cocircuits
+        raise NotImplementedError("cocircuits not implemented")
+
+    def vectors(self):
+        """
+        Return all vectors.
+        """
+        if hasattr(self, "_vectors"):
+            return self._vectors
+        raise NotImplementedError("vectors not implemented")
+        pass
+
+    def covectors(self):
+        """
+        Return all covectors.
+        """
+        if hasattr(self, "_covectors"):
+            return self._covectors
+        raise NotImplementedError("covectors not implemented")
+
+    def convert_to(self, new_type=None):
+        '''
+        Return an oriented matroid of type specified.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: M = OrientedMatroid([[1], [-1], [0]], key='vector')
+            sage: M.convert_to('circuit')
+            Circuit oriented matroid of rank 0
+            sage: M.convert_to()
+            Traceback (most recent call last):
+            ...
+            TypeError: must be given a type to convert to
+        '''
+        from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+        if new_type is None:
+            raise TypeError("must be given a type to convert to")
+        elif new_type in OrientedMatroid.keys:
+            if hasattr(self, new_type + 's'):
+                els = getattr(self, new_type + 's')()
+            else:
+                raise NotImplementedError("no %ss() method found in oriented matroid" % (new_type,))
+            return OrientedMatroid(els,
+                                   key=new_type,
+                                   groundset=self.groundset())
+        else:
+            raise NotImplementedError("type %s not implemented" % (new_type,))
+
+    def dual(self):
+        """
+        Return the dual oriented matroid.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: A = hyperplane_arrangements.braid(3)
+            sage: M = OrientedMatroid(A)
+            sage: m = M.matroid()
+            sage: m.dual()
+            Dual of 'Matroid of rank 2 on 3 elements with 5 flats'
+            sage: m.dual() is m
+            False
+        """
+        pass
+
+    @cached_method
+    def matroid(self):
+        r"""
+        Return the underlying matroid.
+        """
+        pass
```

Comment:
```suggestion
```
Abstract methods also still need a doctest (from one of the concrete implementations).

Reply:
Is there a reason we're removing `pass` here? I guess I'm not understanding when it should be used if it's not used in these circumstances?

Reply:
The docstring actually does the same job as `pass`, but by not having it, it suggests there is no implementation (rather than `pass` which would simply do nothing).